raft/src/main/java/org/apache/kafka/raft/UnattachedState.java

```diff
@@ -39,26 +40,30 @@
  * either through random Fetch requests to the bootstrap servers or through BeginQuorumEpoch
  * request from the leader.
  */
+
 public class UnattachedState implements EpochState {
-    private final int epoch;
-    private final OptionalInt leaderId;
-    private final Set<Integer> voters;
-    private final long electionTimeoutMs;
-    private final Timer electionTimer;
+    private int epoch;
+    private OptionalInt leaderId;
+    private Optional<VotedState> votedState;
+    private Set<Integer> voters;
+    private long electionTimeoutMs;
+    private Timer electionTimer;
```

Comment:
Let's keep these fields finals. You implemented the solution by transitioning to a new unattached state with the voted state set.

Reply:
thanks for that catch